### ISSUE TITLE
How to link against Boost

### ISSUE THREAD

**Issue by OzymandiasTheGreat** [ammarahm-ed/react-native-simple-jsi#17]
Hello.
I'm building a JSI module, I opted to use @mrousavy's template as a base, but I'm also using code
from this repo to export Java methods.
I need to use Boost libraries in my module, but I can't figure out how to link against it.
I figured out how to include Boost headers based on `react-native-vision-camera`'s CMakeLists.txt,
but I can't figure out where Boost binaries are stored and it breaks at the linking step.

Could you help me out here? Boost libraries are very useful for C++ development, so I think this
could be useful for a lot of people going forward.

**Comment by ammarahm-ed** [comment]
Hi, in build.gradle file you will need to download boost library in a downloads folder then extract it into a folder "third-party-ndks", which you can then use to link the library in your project. You can look at projects "react-native-v8" or ", "react-native-reanimated" to know how to do it.

**Comment by OzymandiasTheGreat** [comment]
Thank you, but that doesn't seem to work.
That's exactly what I tried (and a few other libraries using JSI) but as far as I can tell
none of these libraries link against Boost directly.
What they do is download Boost for react native (which seems header only btw)
and include the headers, but since they don't make use of Boost directly they don't link
against it either.
My build fails at link time because I'm using Boost directly but can't find Boost prebuilds
for linking :cry:

**Comment by ammarahm-ed** [comment]
@OzymandiasTheGreat Can you show me the build error or a repo where you are trying to build with boost? I also think it's not headers only since the latest boost library is like 100+ MBS when linked.

**Comment by OzymandiasTheGreat** [comment]
The download link they're using (https://github.com/react-native-community/boost-for-react-native/releases) shows the download as only 11Mb. I think I'm beginning to see where the problem lies.

I double checked build.gradle of a few JSI libraries and according to reanimated's build
script normal boost can be used with react native 0.69. It still uses this header only download
for older versions.

I'm gonna try to build with full boost download according to reanimated's build.gradle and get back to you. Thank you very much for taking the time.

**Comment by ammarahm-ed** [comment]
For react native 0.69+ you need to get boost 1.76 which is around 128Mbs i think

**Comment by OzymandiasTheGreat** [comment]
Finally figured it out. As I noticed earlier, react-native versions <= 0.68 use only Boost headers, which is ok for their purposes because most of Boost is header-only libraries, to begin with. However, if you want to link Boost shared libraries this won't work 'cause they're neither included nor built.

I made a [gist](https://gist.github.com/OzymandiasTheGreat/f188fad84e9e2df1ff262cf84d120e7f) showing how to download and build Boost for Android. On iOS, things are much simpler thanks to [boost-iosx](https://github.com/apotocki/boost-iosx) repo.

Thanks again for your help!
